### PR TITLE
Implemented ops.Where for pandas backend.

### PR DESCRIPTION
**Pull request by DiegoAlbertoTorres**

### CODE REVIEW

**Comment by DiegoAlbertoTorres** [comment]
@cpcloud do you know what these failed tests might be? They live in `ibis.clickhouse.tests.test_functions`, but I doubt I touched anything relevant to those.

**Comment by cpcloud** [comment]
@DiegoAlbertoTorres you need to rebase on top of master

**Comment by cpcloud** [comment]
those tests were fixed in the most recent commit

File: ibis/pandas/execution/tests/test_operations.py [hunk 1]
```diff
@@ -727,3 +727,31 @@ def test_left_binary_op_gb(t, df, op, args):
...
+    result = ibis.where(col_expr > col_expr.mean(), col_expr, 0.0).execute()
+
+    ser = df['plain_int64']
+    expected = ser.where(ser > ser.mean(), other=0.0)
+
+    tm.assert_series_equal(result, expected)
+
+
+def test_where_scalar(t, df):
+    result1 = ibis.where(ibis.literal(True), t['plain_int64'], 3.0).execute()
```
Comment (cpcloud) [inline ibis/pandas/execution/tests/test_operations.py]: FYI you don't need to call literal here, ibis will handle converting python literals into `Literal` nodes

**Comment by cpcloud** [comment]
@DiegoAlbertoTorres if you can rebase then i will merge once this passes.

**Comment by DiegoAlbertoTorres** [comment]
Just rebased.

**Comment by cpcloud** [comment]
I'm going to merge this. Appveyor is flaking out a bit today.

**Comment by cpcloud** [comment]
Oh nice, looks good. Forgot to refresh the page before commenting :)

**Comment by cpcloud** [comment]
thanks @DiegoAlbertoTorres!

**Pull request closed without merging**